playground/backend/internal/code_processing/code_processing.go

```diff
@@ -94,18 +96,19 @@ func Process(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycl
 		if err = processStep(ctxWithTimeout, pipelineId, cacheService, cancelChannel, successChannel, &compileOutput, &compileError, errorChannel, pb.Status_STATUS_COMPILE_ERROR, pb.Status_STATUS_EXECUTING); err != nil {
 			return
 		}
-	case pb.Sdk_SDK_PYTHON:
+		processSuccess(ctx, []byte(""), pipelineId, cacheService, pb.Status_STATUS_EXECUTING)
+	case pb.Sdk_SDK_PYTHON: // No compile step for Python
 		processSuccess(ctx, []byte(""), pipelineId, cacheService, pb.Status_STATUS_EXECUTING)
 	}
 
 	// Run
 	if sdkEnv.ApacheBeamSdk == pb.Sdk_SDK_JAVA {
 		executor = setJavaExecutableFile(lc, pipelineId, cacheService, ctxWithTimeout, executorBuilder, appEnv.WorkingDir())
 	}
-	logger.Infof("%s: Run() ...\n", pipelineId)
-	runCmd := executor.Run(ctxWithTimeout)
+	runCmd := getRunOrTestCmd(&validationResults, &executor, ctxWithTimeout)
```

Comment:
The same question about the references and map.

Reply:
Done